docs/docsite/rst/dev_guide/developing_plugins.rst

```diff
@@ -74,6 +74,62 @@ Plugins that support embedded documentation (see :ref:`ansible-doc` for the list
 Developing particular plugin types
 ==================================
 
+.. _developing_actions:
+
+Action plugins
+--------------
+
+Action plugins are useful when you want to modify data which is provided to any module or filter the data which is returned by the module.
+
+To create an action plugin, create a new class with the Base(ActionBase) class as the parent:
+
+.. code-block:: python
+
+    from ansible.plugins.action import ActionBase
+
+    class ActionModule(ActionBase):
+        pass
+
+
+From there, execute module using ``_execute_module`` method to call original module. After successful execution of module, you can modify module return data.
+
+.. code-block:: python
+
+    module_return = self._execute_module(module_name='<NAME_OF_MODULE>',
+                                         module_args=module_args,
+                                         task_vars=task_vars, tmp=tmp)
+
+The following is an example showing how one can modified return data of Ansible's setup module
+
+.. code-block:: python
+
+    #!/usr/bin/python
+    # Make coding more python3-ish, this is required for contributions to Ansible
+    from __future__ import (absolute_import, division, print_function)
+    __metaclass__ = type
+
+    from ansible.plugins.action import ActionBase
+
+    class ActionModule(ActionBase):
+        def run(self, tmp=None, task_vars=None):
+            super(ActionModule, self).run(tmp, task_vars)
+            module_args = self._task.args.copy()
+            module_return = self._execute_module(module_name='setup',
+                                                 module_args=module_args,
+                                                 task_vars=task_vars, tmp=tmp)
+            ret = dict()
+            if not module_return.get('failed'):
+                for key, value in module_return['ansible_facts'].items():
+                    if key == 'ansible_system':
+                        ret[key] = value
```

Comment:
can i suggest a different example? one that actually shows use of local resources:

timediff action, checks remote clock vs local clock (run date on both machines and compare)

This is not really useful due to network drift but it does show why you would want to use an action plugin.